tests/plugins/testgnupgplugin/tst_gnupgplugin.cpp

```diff
@@ -0,0 +1,225 @@
+ * BSD 3-Clause License, see LICENSE.
+ */
+
+#include <QtTest>
+#include <QSignalSpy>
+
+#include "Crypto/cryptomanager.h"
+#include "Crypto/storedkeyidentifiersrequest.h"
+#include "Crypto/keypairgenerationparameters.h"
+#include "Crypto/generatekeyrequest.h"
+#include "Crypto/signrequest.h"
+#include "Crypto/verifyrequest.h"
+#include "Crypto/key.h"
+#include "Crypto/result.h"
+
+using namespace Sailfish::Crypto;
+
+#define OPENPGP_PLUGIN QStringLiteral("org.sailfishos.crypto.plugin.gnupg.openpgp")
+
+// Cannot use waitForFinished() for some replies, as ui flows require user interaction / event handling.
+#define WAIT_FOR_FINISHED_WITHOUT_BLOCKING(request)                         \
+    do {                                                                    \
+        int maxWait = 1000000;                                                \
+        while (request.status() != (int)Request::Finished && maxWait > 0) { \
+            QTest::qWait(100);                                              \
+            maxWait -= 100;                                                 \
+        }                                                                   \
+    } while (0)
+#define SHORT_WAIT_FOR_FINISHED_WITHOUT_BLOCKING(request)                   \
+    do {                                                                    \
+        int maxWait = 1000000;                                                \
+        while (request.status() != (int)Request::Finished && maxWait > 0) { \
+            QTest::qWait(1);                                                \
+            maxWait -= 1;                                                   \
+        }                                                                   \
+    } while (0)
+#define LONG_WAIT_FOR_FINISHED_WITHOUT_BLOCKING(request)                    \
+    do {                                                                    \
+        int maxWait = 3000000;                                                \
+        while (request.status() != (int)Request::Finished && maxWait > 0) { \
+            QTest::qWait(100);                                              \
+            maxWait -= 100;                                                 \
+        }                                                                   \
+    } while (0)
+
+class tst_gnupgplugin : public QObject
+{
+    Q_OBJECT
+
+public slots:
+    void init();
+    void cleanup();
+
+private slots:
+    void signVerify();
+    void signVerify_data();
+    void storedKeyIdentifiers();
+
+private:
+    CryptoManager cm;
+};
+
+void tst_gnupgplugin::init()
+{
+}
+
+void tst_gnupgplugin::cleanup()
+{
+}
+
+void tst_gnupgplugin::storedKeyIdentifiers()
+{
+    StoredKeyIdentifiersRequest req;
+
+    req.setManager(&cm);
+    QSignalSpy reqstat(&req, &StoredKeyIdentifiersRequest::statusChanged);
+    QSignalSpy reqiden(&req, &StoredKeyIdentifiersRequest::identifiersChanged);
+
+    // req.setCryptoPluginName(OPENPGP_PLUGIN);
```

Comment:
With the refactor, you need to do: setStoragePluginName(OPENPGP_PLUGIN); and also that plugin must also implement the EncryptedStoragePlugin interface (as well as the CryptoPlugin interface) -- see e.g. the SqlCipher plugin for example.

Reply:
Thanks, I've added implementation for the EncryptedStorage API.